src/preferences/dialog/dlgprefmixer.cpp

```diff
@@ -779,72 +779,84 @@ void DlgPrefMixer::slotUpdate() {
-        pXfScene->addLine(
-                QLineF(0, i * (sizeY / kGrindXLines), sizeX, i * (sizeY / kGrindXLines)), gridPen);
+    const int kGridHLines = 3;
+    const int kGridVLines = 5;
+    const int kGridHDist = static_cast<int>(sceneH / (kGridHLines + 1));
+    const int kGridVDist = static_cast<int>(sceneW / (kGridVLines + 1));
+    for (int i = 1; i <= kGridHLines; i++) {
+        m_pxfScene->addLine(QLineF(0, i * kGridHDist, sceneW, i * kGridHDist), gridPen);
     }
-    for (int i = 1; i < kGrindYLines; i++) {
-        pXfScene->addLine(
-                QLineF(i * (sizeX / kGrindYLines), 0, i * (sizeX / kGrindYLines), sizeY), gridPen);
+    for (int i = 1; i <= kGridVLines; i++) {
+        m_pxfScene->addLine(QLineF(i * kGridVDist, 0, i * kGridVDist, sceneH), gridPen);
     }
 
-    // Draw graph lines
-    QPointF pointTotal, point1, point2;
-    QPointF pointTotalPrev, point1Prev, point2Prev;
-    int pointCount = sizeX - 4;
-    // reduced by 2 x 1 for border + 2 x 1 for inner distance to border
-    double xfadeStep = 2. / (pointCount - 1);
-    for (int i = 0; i < pointCount; i++) {
-        CSAMPLE_GAIN gain1, gain2;
+    // Draw gain curves
+    QPolygonF polylineTotal;
+    QPolygonF polylineL;
+    QPolygonF polylineR;
+    // Required to have symmetrical curves not overdraw at the right
+    int pointCount = sceneW - 2;
+    double xfadeStep = 2. / (pointCount);
+    for (int i = 0; i <= pointCount; i++) {
+        CSAMPLE_GAIN gainL, gainR;
         EngineXfader::getXfadeGains((-1. + (xfadeStep * i)),
                 m_transform,
                 m_cal,
                 m_xFaderMode,
                 checkBoxReverse->isChecked(),
-                &gain1,
-                &gain2);
+                &gainL,
+                &gainR);
 
-        double gain = sqrt(gain1 * gain1 + gain2 * gain2);
+        double gainTotal = sqrt(gainL * gainL + gainR * gainR);
         // scale for graph
-        gain1 *= 0.71f;
-        gain2 *= 0.71f;
-        gain *= 0.71f;
-
-        // draw it
-        pointTotal = QPointF(i + 1, (1. - gain) * (sizeY)-3);
-        point1 = QPointF(i + 1, (1. - gain1) * (sizeY)-3);
-        point2 = QPointF(i + 1, (1. - gain2) * (sizeY)-3);
-
-        if (i > 0) {
-            pXfScene->addLine(QLineF(pointTotal, pointTotalPrev), QPen(Qt::red));
-            pXfScene->addLine(QLineF(point1, point1Prev), graphLinePen);
-            pXfScene->addLine(QLineF(point2, point2Prev), graphLinePen);
-        }
+        gainL *= 0.71f;
+        gainR *= 0.71f;
+        gainTotal *= 0.71f;
 
-        // Save old values
-        pointTotalPrev = pointTotal;
-        point1Prev = point1;
-        point2Prev = point2;
+        polylineTotal.append(QPointF(i + 1, (1. - gainTotal) * (sceneH)-3));
+        polylineL.append(QPointF(i + 1, (1. - gainL) * (sceneH)-3));
+        polylineR.append(QPointF(i + 1, (1. - gainR) * (sceneH)-3));
     }
 
-    graphicsViewXfader->setScene(pXfScene);
+    QPainterPath pathTotal;
```

Comment:
missing include...

Reply:
builds fine on Ubu 20.04 locally and on macOS, so only  22.04 and Win need the include?
"Verrry, verrry strange!"

Reply:
Well, probably just a transient include from somewhere else that got removed later... thats why you always include what you use even if some of the previous includes _may_ already have done so.